### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by ivanaitzliddat**
Update till session related commands

### CODE REVIEW

**Review by froststein — APPROVED** [review]
Looks good to merge

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -172,7 +176,7 @@ to write all the data stored in the `Profile` component into the save file witho
 The `Parser` component consists of the `Parser` class, `ParserUtils` class, `ParserErrors` class,
 as well as the `CommandParser` class and its subclasses.<br>
 * The `Parser` class provides methods to parse commands from the user and
-  return a `Command` object representing an instruction that the user has for SplitLah.<br>
+  returns a `Command` object representing an instruction that the user has for SplitLah.<br>
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:179]: "to parse ... and to return", nothing wrong here, should not be edited.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -186,11 +190,10 @@ The general workflow of the `Parser` component is as follows:
...
 3. The `XYZCommandParser` object then uses parse methods from `ParserUtils` class to extract all the
    arguments from the user input.
-   * Each of these parse methods in `ParserUtils` class then calls other utility methods within the class 
+   * Each of these parse methods in `ParserUtils` class calls other utility methods within the class 
      to return a parsed value.
-4. All relevant arguments that are parsed are then used to create a new `XYZCommand `object to be
-   returned to the `Parser` class.
-5. The created `XYZCommand` object is then returned to the `SplitLah` object to be run.
+4. All relevant arguments that are parsed are used to create a new `XYZCommand `object to return to the `Parser` class.
+5. The created `XYZCommand` object is returned to the `SplitLah` object to run.
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:196]: Consider keeping the word then to maintain the flow of the discussion.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -17,7 +18,7 @@ title: Developer Guide
     * [Parser Component](#parser-component)
     * [Command Component](#command-component)
 * [Implementation](#implementation)
-  * [Parser implementation](#parsing-of-commands)
+  * [Parser of commands](#parsing-of-commands)
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:21]: Maybe "Parsing of commands" instead would be more representative.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -94,30 +97,30 @@ The diagram depicts a scenario where a user attempts to create a session.
 ![SplitLah Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/SplitLahComponent.drawio.png)
 <br>
 The `SplitLah` component is the application's main class. Its job is to initialize an instance of `Manager` when the
-application starts. After initialization, it proceeds to run a loop which prompts the user for a 
-command. When it receives a command from the user, it invokes the `Parser` and retrieve the command for SplitLah
-to run. Upon using the `Exit` command, SplitLah exits from the command loop and end the application.
+application starts. After initialization, it proceeds to run a loop which prompts the user for an input.
+When it receives an input from the user, it invokes the `Parser` and retrieve a command corresponding to the input for SplitLah
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:101]: Consider correcting the grammatical error here: It should be "invokes ... and retrieves ...".

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -240,10 +244,10 @@ the Command component when any user input is provided to SplitLah.
    `session /create /n Class Outing /d 15-03-2022 /pl Alice Bob` is parsed separately as
    `session /create` and `/n Class Outing /d 15-03-2022 /pl Alice Bob`.
    1. If the _command type_ is invalid, the method `Parser#getCommandType` returns null to `Parser` class.
-      An `InvalidCommand` object is then created and returned to `SplitLah`.
-   2. `Parser` class then validates the _command type_ and the _remaining arguments_ with
+      As a result, an `InvalidCommand` object is created and returned to `SplitLah`.
+   2. Else, `Parser` class validates the _command type_ and the _remaining arguments_ with
       `Parser#checkIfCommandIsValid`. If either the _command type_ or the _remaining arguments_ are invalid, an error
-      message is returned by the method, which is then used to return an `InvalidCommand` object to `SplitLah`.
+      message is returned by the method, which returns an `InvalidCommand` object to `SplitLah`.
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:250]: This change is a little weird, consider rephrasing for better flow. What is referred to by "which"?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -263,8 +267,8 @@ the Command component when any user input is provided to SplitLah.
    * For example, `ParserUtils#parseName` returns a `String` object representing a name.
 7. After all necessary information is parsed, `XYZCommandParser` instantiates a new `XYZCommand` object with
    all the parsed information.
-8. The `XYZCommand` object is then returned from `XYZCommandParser` to `Parser` class,
-   and finally back to `SplitLah` to be run.
+8. Following that, The `XYZCommand` object is returned from `XYZCommandParser` to `Parser` class,
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:270]: Consider correcting the case used to lower case for "The".

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -428,12 +432,12 @@ The general workflow of the `session /summary` command is as follows:
      to be collected between all persons. Each of such matches is referred to as a transaction.
      The matching process is repeated until no more transactions can be made,
      i.e. all debts are paid and all debts are collected.
-   * A `String` object containing information regarding all transactions that have to be made is then returned.
    * If no transactions are required to be made, a message explaining that no transactions are required to be made
      is returned instead.
+   * Else, a `String` object containing information regarding all transactions that have to be made is returned.
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:437]: Consider removing "instead" if you were to shift the previous clause down.

**Review by warrencxw — CHANGES_REQUESTED** [review]
Added many suggestions, please refer to the comments and edit as necessary.

**Review by froststein — CHANGES_REQUESTED** [review]
Requested change

**Review by warrencxw — APPROVED** [review]
Looks good to merge.

**Review by froststein — APPROVED** [review]
Looks good to merge

**Pull request merged**